docs/src/modules/components/AppNavDrawerItem.js

```diff
@@ -256,15 +257,15 @@ export default function AppNavDrawerItem(props) {
     comingSoon,
     linkProps,
     onClick,
-    openImmediately,
+    initiallyExpanded = false,
+    expandable = false,
     plan = 'community',
     subheader,
     title,
     topLevel = false,
     ...other
   } = props;
-  const expandable = openImmediately != null;
```

Comment:
There was a problem with `expandable` - it was `true` when `subheader === true`. This was wrong and resulted with:
- Unnecessary `Collapse` wrapper, even though subheaders are never collapsible
- Wrong subheader positioning in some cases:

  <img width="301" src="https://github.com/mui/material-ui/assets/13808724/bd23678f-b4be-4c4a-917a-7b9042789303">

I've added the `expandable` prop instead, which is only `true` if the page has children and is not a subheader:

<img width="299" src="https://github.com/mui/material-ui/assets/13808724/6bd38919-f56e-4b32-bb6a-4d513f368341">